src/App.tsx

```diff
@@ -1,14 +1,83 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [filter, setFilter] = useState('all');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeTodo, setActiveTodo] = useState<Todo | undefined>(undefined);
+  const [query, setQuery] = useState('');
+  const [loadingTodos, setLoadingTodos] = useState(true);
+
+  useEffect(() => {
+    setLoadingTodos(true);
+
+    switch (filter) {
+      case 'all':
+        getTodos()
```

Comment:
No need to fetch todos every time filter changes
Fetch them once. Then just save in the common variable (const) filtered todos. It's better to use  the `useMemo` hook to memorize value and recompute filtered todos only on filter or query change

Reply:
Okay, for this task this will work as we have static todos. We can change it later to update todos periodically or something if needed. Also will save some resources for Mate Academy =)